### ISSUE TITLE
In Cluster configuration broken

### ISSUE THREAD

**Issue by ctron** [clux/kube-rs#251]
For me the "in cluster" configuration seems broken:

I am doing:
~~~rust
let client = Client::try_default().await?;
~~~

And that fails later with:

~~~
[2020-06-17T16:07:00Z INFO  ditto_operator] Starting operator...
[2020-06-17T16:07:00Z TRACE kube::runtime::reflector] Resetting Ditto
[2020-06-17T16:07:00Z TRACE hyper::client::pool] checkout waiting for idle connection: ("https", 172.30.0.1)
[2020-06-17T16:07:00Z DEBUG reqwest::connect] starting new connection: https://172.30.0.1/
[2020-06-17T16:07:00Z TRACE hyper::client::connect::http] Http::connect; scheme=Some("https"), host=Some("172.30.0.1"), port=None
[2020-06-17T16:07:00Z DEBUG hyper::client::connect::http] connecting to 172.30.0.1:443
[2020-06-17T16:07:00Z TRACE mio::poll] registering with poller
[2020-06-17T16:07:00Z DEBUG hyper::client::connect::http] connected to 172.30.0.1:443
[2020-06-17T16:07:00Z TRACE mio::poll] deregistering handle with poller
[2020-06-17T16:07:00Z TRACE hyper::client::pool] checkout dropped for ("https", 172.30.0.1)
Error: ReqwestError: error sending request for url (https://172.30.0.1/apis/iot.eclipse.org/v1alpha1/namespaces/ditto/dittos?): error trying to connect: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:1915: (self signed certificate in certificate chain)

Caused by:
    0: error sending request for url (https://172.30.0.1/apis/iot.eclipse.org/v1alpha1/namespaces/ditto/dittos?): error trying to connect: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:1915: (self signed certificate in certificate chain)
    1: error trying to connect: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:1915: (self signed certificate in certificate chain)
    2: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:1915: (self signed certificate in certificate chain)
    3: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:1915:
~~~

However, doing a simple `curl` seems to work just fine:

~~~
$ curl https://172.30.0.1/apis/iot.eclipse.org/v1alpha1/namespaces/ditto/dittos? --cacert /run/secrets/kubernetes.io/serviceaccount/ca.crt --oauth2-bearer $(cat /run/secrets/kubernetes.io/serviceaccount/token)  
{"apiVersion":"iot.eclipse.org/v1alpha1","items":[{"apiVersion":"iot.eclipse.org/v1alpha1","kind":"Ditto","metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"iot.eclipse.org/v1alpha1\",\"kind\":\"Ditto\",\"metadata\":{\"annotations\":{},\"name\":\"test\",\"namespace\":\"ditto\"},\"spec\":{}}\n"},"creationTimestamp":"2020-05-18T16:43:37Z","generation":3,"name":"test","namespace":"ditto","resourceVersion":"197353246","selfLink":"/apis/iot.eclipse.org/v1alpha1/namespaces/ditto/dittos/test","uid":"a2abc8d4-3954-4177-b806-674896b7ec6c"},"spec":{"mongoDb":{"host":"mongodb"}},"status":{"phase":"Active"}}],"kind":"DittoList","metadata":{"continue":"","resourceVersion":"209819082","selfLink":"/apis/iot.eclipse.org/v1alpha1/namespaces/ditto/dittos"}}
~~~

**Comment by clux** [comment]
hm, certificate issue on in-cluster? that's almost certainly a bug.

not sure what's the problem here yet, but will have a go at reproducing later.

**Comment by ctron** [comment]
I think I see the problem. The `ca.crt` file contains a chain of certificates, while the configuration only reads one.

**Comment by ctron** [comment]
Yes, that was it. I added some logic to split the PEM and add all certificates in that bundle, and that made it work. I can create a PR if your like?!

**Comment by clux** [comment]
Thanks again. This was super helpful!

Have released your pr in a patch release `kube = 0.35.1`

**Comment by ctron** [comment]
Thanks for the quick merge. And for the service release :+1:
